scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java

```diff
@@ -1197,36 +1279,41 @@ private void build() {
 
     private Boolean isExecutionCalendarGIDefined(String jobDescriptor) {
         Document dom = XMLParser.parse(jobDescriptor);
-        Boolean exists = false;
-        Boolean executionCalendarDefined = false;
+
         NodeList genericInfo = dom.getElementsByTagName("genericInformation");
         // check if the job has genericInformation or not
         if (genericInfo != null && genericInfo.getLength() > 0) {
-            // get the first item
-            Node root = genericInfo.item(0);
-            NodeList list = root.getChildNodes();
-            for (int i = 0; i < list.getLength(); i++) {
-                Node node = list.item(i);
-                if (node.getNodeName().equals("info") && node.hasAttributes()) {
-                    NamedNodeMap attributes = node.getAttributes();
-                    for (int j = 0; j < attributes.getLength(); j++) {
-                        Node attribute = attributes.item(j);
-                        if (attribute.getNodeType() == Node.ATTRIBUTE_NODE && attribute.getNodeName().equals("name") &&
-                            attribute.getNodeValue().equalsIgnoreCase("execution_calendars")) {
-                            exists = true;
-                        }
-                        if (isAttributeExecCalendarValueDefined(attribute, "value") && exists) {
-                            executionCalendarDefined = true;
-                            if (!attribute.getNodeValue().isEmpty())
-                                isExecCalendarValueNull = false;
-                        }
+            // search calendar GI in the first item
+            return searchCalendarGI(genericInfo.item(0));
+        } else
+            return false;
+    }
 
+    private Boolean searchCalendarGI(Node genericInfoNode) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 22 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)